osgi.specs/docbook/core/058/service.resolver.xml

```diff
@@ -1415,6 +1416,144 @@ public List&lt;Capability&gt; findProviders( Requirement requirement) {
+        often modeling non-OSGi artifacts (like Maven dependencies) as
+        resources with requirements and capabilities.</para>
+      </listitem>
+
+      <listitem>
+        <para><emphasis>IDEs and Development Tools</emphasis> - Integrated
+        development environments need to analyze bundle dependencies and
+        predict resolution outcomes without running a full OSGi
+        framework.</para>
+      </listitem>
+
+      <listitem>
+        <para><emphasis>Provisioning Tools</emphasis> - Tools that operate on
+        repositories to download and provision bundles often need resolution
+        capabilities but do not themselves run within OSGi.</para>
+      </listitem>
+
+      <listitem>
+        <para><emphasis>Feature Launchers</emphasis> - Tools that construct
+        and launch OSGi frameworks based on feature definitions (see <xref
+        linkend="service.featurelauncher"/>) need to resolve features before
+        the framework starts.</para>
+      </listitem>
+    </itemizedlist>
+
+    <para>For these use cases, obtaining a resolver by referencing a specific
+    implementation class creates tight coupling and prevents the use of
+    alternative resolver implementations. To address this, resolver
+    implementations can be discovered using the Java Service Provider
+    mechanism.</para>
+
+    <section>
+      <title>The Resolver SPI</title>
+
+      <para>The <xref linkend="org.osgi.service.resolver.Resolver"
+      xrefstyle="hyperlink"/> interface serves as the Service Provider
+      Interface (SPI) for resolver implementations. Resolver implementation
+      providers must implement this interface and register it using the Java
+      Service Provider mechanism.</para>
+
+      <para>A resolver implementation JAR must contain the following
+      resource:</para>
+
+      <programlisting>META-INF/services/org.osgi.service.resolver.Resolver</programlisting>
+
+      <para>This UTF-8 encoded resource must contain the fully qualified name
+      of the Resolver implementation class. The format follows the standard
+      Java Service Provider configuration format: each line contains either a
+      class name or a comment (lines starting with '#'). Space and tab
+      characters, including blank lines, must be ignored.</para>
+
+      <para>For example, if the <code>com.acme.osgi</code> package includes
+      a resolver implementation class <code>com.acme.osgi.ResolverImpl</code>,
+      it should have the following resource:</para>
+
+      <programlisting>META-INF/services/org.osgi.service.resolver.Resolver</programlisting>
+
+      <para>With the contents:</para>
+
+      <programlisting># ACME Resolver Implementation
+com.acme.osgi.ResolverImpl</programlisting>
+
+      <para>The Resolver implementation class must have a public, no-argument
+      constructor. Each instance is independent and can be used to perform
+      resolution operations.</para>
+    </section>
```

Comment:
I would model this after other sections where we spec requirements on a `META-INF/services` resource.  For example, like https://docs.osgi.org/specification/osgi.core/8.0.0/framework.connect.html#framework.connect.service.provider